kaldb/src/test/java/com/slack/kaldb/chunkrollover/DiskOrMessageCountBasedRolloverStrategyTest.java

```diff
@@ -0,0 +1,219 @@
+      offset++;
+      Thread.sleep(DiskOrMessageCountBasedRolloverStrategy.DIRECTORY_SIZE_EXECUTOR_PERIOD_MS);
+      if (chunkManager.getActiveChunk() != null) {
+        chunkManager.getActiveChunk().commit();
+      }
+      // this doesn't work because the next active chunk gets assigned only on next message add
+      //        await()
+      //            .untilAsserted(
+      //                () -> assertThat(getValue(LIVE_BYTES_DIR, metricsRegistry)).isEqualTo(0));
+      if (shouldCheckOnNextMessage) {
+        assertThat(getValue(LIVE_BYTES_DIR, metricsRegistry)).isEqualTo(0);
+      }
+      shouldCheckOnNextMessage = getValue(LIVE_BYTES_DIR, metricsRegistry) > MAX_BYTES_PER_CHUNK;
+    }
+    assertThat(getCount(RollOverChunkTask.ROLLOVERS_INITIATED, metricsRegistry)).isEqualTo(2);
+    assertThat(getCount(RollOverChunkTask.ROLLOVERS_FAILED, metricsRegistry)).isEqualTo(0);
+    assertThat(getCount(RollOverChunkTask.ROLLOVERS_COMPLETED, metricsRegistry)).isEqualTo(2);
+
+    KaldbSearch.SearchRequest.Builder searchRequestBuilder = KaldbSearch.SearchRequest.newBuilder();
+
+    final long chunk1StartTimeMs = startTime.toEpochMilli();
+    final long chunk1EndTimeMs = chunk1StartTimeMs + (totalMessages * 1000);
+    KaldbSearch.SearchResult response =
+        kaldbLocalQueryService.doSearch(
+            searchRequestBuilder
+                .setDataset(MessageUtil.TEST_DATASET_NAME)
+                .setQueryString("Message1")
+                .setStartTimeEpochMs(chunk1StartTimeMs)
+                .setEndTimeEpochMs(chunk1EndTimeMs)
+                .setHowMany(10)
+                .setBucketCount(2)
+                .build());
+
+    assertThat(response.getHitsCount()).isEqualTo(1);
+    assertThat(response.getTookMicros()).isNotZero();
+    assertThat(response.getTotalCount()).isEqualTo(1);
+    assertThat(response.getFailedNodes()).isZero();
+    assertThat(response.getTotalNodes()).isEqualTo(1);
+    assertThat(response.getTotalSnapshots()).isEqualTo(3);
+    assertThat(response.getSnapshotsWithReplicas()).isEqualTo(3);
+  }
+
+  @Test
+  public void testChunkRollOver() {
+    ChunkRollOverStrategy chunkRollOverStrategy =
+        new DiskOrMessageCountBasedRolloverStrategy(metricsRegistry, 1000, 2000);
+
+    assertThat(chunkRollOverStrategy.shouldRollOver(1, 1)).isFalse();
+    assertThat(chunkRollOverStrategy.shouldRollOver(-1, -1)).isFalse();
+    assertThat(chunkRollOverStrategy.shouldRollOver(0, 0)).isFalse();
+    assertThat(chunkRollOverStrategy.shouldRollOver(100, 100)).isFalse();
+    assertThat(chunkRollOverStrategy.shouldRollOver(1000, 1)).isFalse();
+    assertThat(chunkRollOverStrategy.shouldRollOver(1001, 1)).isFalse();
+
+    assertThat(chunkRollOverStrategy.shouldRollOver(100, 2000)).isTrue();
+    assertThat(chunkRollOverStrategy.shouldRollOver(100, 2001)).isTrue();
+    assertThat(chunkRollOverStrategy.shouldRollOver(1001, 2001)).isTrue();
+  }
```

Comment:
This looks great! Please add a complete test using message count rollover like the unit test `testDiskBasedRolloverWithMultipleChunks`.